### PR TITLE
Implement Set mentor availability endpoint (Admin) #28

### LINKED ISSUE #28: Implement Set mentor availability endpoint (Admin)
**Description:**
This issue involves implementing an API to get all mentors. The endpoint should allow Admin to make a PUT request to `/api/admin/mentor/{{mentorId}}/availability`.

Example mark mentor as available
```
{
    "mentor_id": 1,
    "created_at": "2023-06-30T10:00:00Z",
    "updated_at": "2023-06-30T14:30:00Z",
    "state": "APPROVED",
    "category": "Technology",
    "application": {
        "country": "",
        "areasOfExpertise": "",
        "expectationsFromMentees": "",
        "mentoringPhilosophy": "",
        "commitmentToProgram": false,
        "previousExperienceAsMentor": false,
        "reasonForBeingMentor": "",
        "cvLink": ""
    },
    "availability": true,
    "profile": {
        "created_at": "2023-06-29T09:15:00Z",
        "updated_at": "2023-06-30T16:45:00Z",
        "primary_email": "mentor@example.com",
        "contact_email": "mentor_contact@example.com",
        "first_name": "John",
        "last_name": "Doe",
        "image_url": "https://example.com/mentor_profile_image.jpg",
        "linkedin_url": "https://www.linkedin.com/in/johndoe",
        "type": "DEFAULT",
        "uuid": "12345678-1234-5678-1234-567812345678"
    }
}
```

Example mark mentor as not available
```
{
    "mentor_id": 1,
    "created_at": "2023-06-30T10:00:00Z",
    "updated_at": "2023-06-30T14:30:00Z",
    "state": "APPROVED",
    "category": "Technology",
    "application": {
        "country": "",
        "areasOfExpertise": "",
        "expectationsFromMentees": "",
        "mentoringPhilosophy": "",
        "commitmentToProgram": false,
        "previousExperienceAsMentor": false,
        "reasonForBeingMentor": "",
        "cvLink": ""
    },
    "availability": false,
    "profile": {
        "created_at": "2023-06-29T09:15:00Z",
        "updated_at": "2023-06-30T16:45:00Z",
        "primary_email": "mentor@example.com",
        "contact_email": "mentor_contact@example.com",
        "first_name": "John",
        "last_name": "Doe",
        "image_url": "https://example.com/mentor_profile_image.jpg",
        "linkedin_url": "https://www.linkedin.com/in/johndoe",
        "type": "DEFAULT",
        "uuid": "12345678-1234-5678-1234-567812345678"
    }
}
```
**Tasks:**
1. Create a controller for the `/controllers/mentor` endpoint in the backend (create the route mentor if not created).
2. Parse and validate the request body to ensure it matches the expected format.
3. Implement appropriate error handling and response status codes for different scenarios (e.g., validation errors, database errors).
5. Write unit tests to validate the functionality and correctness of the endpoint.

API documentation: https://documenter.getpostman.com/view/27421496/2s93m1a4ac#8744a3ee-970f-489a-853d-8b23fdee8de3

ER diagram: https://drive.google.com/file/d/11KMgdNu2mSAm0Ner8UsSPQpZJS8QNqYc/view 

**Acceptance Criteria:**
- The profile API endpoint `/controllers/mentor` is implemented via a PUT request.
- The request body is properly parsed and validated for the required format.
- Appropriate error handling is implemented, providing meaningful error messages and correct response status codes.
- Unit tests are written to validate the functionality and correctness of the endpoint.

**Additional Information:**
No

**Related Dependencies or References:**
No

### PR DESCRIPTION
**Pull request by sathudeva7**
## Purpose

The purpose of this PR is to fix #28 

## Goals

This issue involves implementing an API to get all mentors. The endpoint should allow Admin to make a PUT request to /api/admin/mentor/{{mentorId}}/availability.

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/controllers/admin/mentor.controller.ts [hunk 1]
```diff
@@ -105,3 +106,33 @@ export const getAllMentorEmails = async (
...
+  try {
+    const user = req.user as Profile
+    const { availability } = req.body
+    const { mentorId } = req.params
+
+    if (user.type !== ProfileTypes.ADMIN) {
+      return res.status(403).json({ message: 'Only Admins are allowed' })
+    }
+
+    const { mentor, statusCode, message } = await updateAvailability(
```
Comment (anjula-sack) [inline src/controllers/admin/mentor.controller.ts:123]: Here we can use the existing service `updateAvailability` in `src/services/mentor.service.ts`

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/controllers/admin/mentor.controller.ts (hunk 1 above)
Comment (sathudeva7) [reply src/controllers/admin/mentor.controller.ts:123]: But exisiting function require user profile as paramter 
```
export const updateAvailability = async (
  user: Profile,
  availability: boolean
): Promise<{ statusCode: number; updatedMentorApplication: Mentor }> => {
```. Here we have mentor Id only

Comment (anjula-sack) [reply src/controllers/admin/mentor.controller.ts:123]: can you update that(updateAvailability) to be the mentorId as well? Both are the same. @sathudeva7

**Comment by anjula-sack** [comment]
Any updates? @sathudeva7

**Comment by sathudeva7** [comment]
> Any updates? @sathudeva7

pushed

File: src/routes/admin/mentor/mentor.route.test.ts [hunk 2]
```diff
@@ -117,4 +120,44 @@ describe('Admin mentor routes', () => {
...
+      const mentor = response.body.updatedMentorApplication
+
+      expect(mentor).toHaveProperty('availability', availability)
+    }
+  )
+
+  it.each([true, false])(
+    'should only allow admins to update the mentor availability',
+    async (availability) => {
+      console.log(mentorProfileId)
```
Comment (anjula-sack) [inline src/routes/admin/mentor/mentor.route.test.ts:141]:
```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Review by anjula-sack — APPROVED** [review]
Thank you! @sathudeva7

**Pull request merged**